crates/compiler/src/lib.rs

```diff
@@ -58,6 +37,34 @@ fn panic_error(e: Box<dyn std::any::Any>) -> anyhow::Error {
     }
 }
 
+#[derive(Debug, PartialEq)]
+struct SierraVersion(u64, u64, u64);
+
+/// Parse Sierra version from the JSON representation of the program.
+///
+/// Sierra programs contain the version number in two possible formats.
+/// For pre-1.0-rc0 Cairo versions the program contains the Sierra version
+/// "0.1.0" as a shortstring in its first Felt (0x302e312e30 = "0.1.0").
+/// For all subsequent versions the version number is the first three felts
+/// representing the three parts of a semantic version number.
+fn parse_sierra_version(program: &serde_json::value::RawValue) -> anyhow::Result<SierraVersion> {
+    let felts: Vec<Felt> =
+        serde_json::from_str(program.get()).context("Deserializing Sierra program felts")?;
+    let parts = &felts[..3];
+
+    const VERSION_0_1_0_AS_SHORTSTRING: Felt = felt!("0x302e312e30");
+
+    match parts {
+        [VERSION_0_1_0_AS_SHORTSTRING, ..] => Ok(SierraVersion(0, 1, 0)),
+        [a, b, c] => {
```

Comment:
Or do a length check before slicing?

```suggestion
    const VERSION_0_1_0_AS_SHORTSTRING: Felt = felt!("0x302e312e30");

    match &felts {
        [VERSION_0_1_0_AS_SHORTSTRING, ..] => Ok(SierraVersion(0, 1, 0)),
        [a, b, c, ...] => {
```

Reply:
Yeah, that's a good point.